packages/core/src/errors/InruptError.ts

```diff
@@ -21,60 +21,107 @@
- *  Then we can either import that implementation in this code, or just import
- *  the types when needed here, e.g.:
- import type { NamedNode } from "rdf-js";
- */
-import { NamedNode } from "n3";
-
-import { VocabTerm } from "@inrupt/solid-common-vocab";
+import { VocabTerm, NamedNode } from "@inrupt/solid-common-vocab";
 
 export default class InruptError extends Error {
-  public readonly statusCode?: number;
-
-  public readonly statusText?: string;
+  // For the common case of HTTP errors, we can store the HTTP response to allow
+  // consumers of this error instance to access it directly.
+  // NOTE: we specifically stipulate that we expect the HTTP response to be an
+  // error response!
+  private httpErrorResponse?: Response & { ok: false };
 
   constructor(
     messageOrIri: string | NamedNode | VocabTerm,
     messageParams?: string[],
-    httpResponse?: Response & { ok: false },
-    appendHttpDetails = true,
     appendErrorIri = true
   ) {
     super(
-      InruptError.appendHttpResponseDetails(
-        typeof messageOrIri === "string"
-          ? InruptError.substituteParams(messageOrIri, messageParams)
-          : InruptError.appendErrorIri(
-              InruptError.lookupErrorIri(messageOrIri, messageParams),
-              messageOrIri as NamedNode,
-              appendErrorIri
-            ),
-        httpResponse,
-        appendHttpDetails
-      )
+      typeof messageOrIri === "string"
+        ? InruptError.substituteParams(messageOrIri, messageParams)
+        : InruptError.appendErrorIri(
+            InruptError.lookupErrorIri(messageOrIri, messageParams),
+            messageOrIri as NamedNode,
+            appendErrorIri
+          )
+    );
+  }
+
+  /**
+   * Allows us provide a HTTP response, and to specify if we want details from
+   * that response to be appended to our error message string. We also preserve
+   * specific state from that response object to allow consumers of this error
+   * to directly access those if they wish.
+   *
+   * @param httpErrorResponse
+   * @param appendHttpDetails
+   */
+  public httpResponse(
+    httpErrorResponse: Response & { ok: false },
+    appendHttpDetails = true
+  ): InruptError {
+    this.message = InruptError.appendHttpResponseDetails(
+      this.message,
+      httpErrorResponse,
+      appendHttpDetails
     );
 
-    if (typeof httpResponse !== "undefined") {
-      this.statusCode = httpResponse.status;
-      this.statusText = httpResponse.statusText;
+    // Preserve the HTTP response (in case the consumer of this error wishes to
+    // it's details directly themselves).
+    this.httpErrorResponse = httpErrorResponse;
+
+    return this;
+  }
+
+  hasHttpResponse(): boolean {
+    return this.httpErrorResponse !== undefined;
+  }
+
+  getHttpResponse(): (Response & { ok: false }) | undefined {
+    return this.httpErrorResponse;
+  }
+
+  getHttpStatusCode(): number {
+    if (!this.hasHttpResponse()) {
+      throw new InruptError(
```

Comment:
If this really is an error if there is no HTTP Status code it should have been left in the constructor (or else use a builder pattern to create the Error). Any client code that is catching this error will have to _always_ have to try can catch these if it calles this function (which can get clunky).

Reply:
Yeah, I was uncomfortable having any method in this class throw an error, but just wanted to start off super-defensive. But yeah, factoring out the HTTP-awareness from this class completely would tidy this up too.